release-tools/metabuild.sh

```diff
@@ -0,0 +1,100 @@
+#!/bin/sh
+
+D=$1
+MAKEBSD=$2
+CFLAGS_HARDCODED=$3
+
+OUT=Makefile
+
+# Set up directories
+cd ${D}
+SUBDIR_DEPTH=`${MAKEBSD} -V SUBDIR_DEPTH`
+
+# Set up *-config.h so that we don't have missing headers
+rm -f ${SUBDIR_DEPTH}/cpusupport-config.h
+touch ${SUBDIR_DEPTH}/cpusupport-config.h
+
+copyvar() {
+	var=$1
+	printf "$var=$(${MAKEBSD} -V $var)\n" >> $OUT
+}
+
+add_makefile_prog() {
+	if [ `${MAKEBSD} -V NOINST` ]; then
+		cat ${SUBDIR_DEPTH}/Makefile.prog |		\
+		    perl -0pe 's/(install:.*?)\n\n//s' >> $OUT
+	else
+		cat ${SUBDIR_DEPTH}/Makefile.prog >> $OUT
+	fi
+}
+
+add_object_files() {
+	# Set up useful variables
+	OBJ=$(${MAKEBSD} -V SRCS |				\
+	    sed -e 's| cpusupport-config.h||' |			\
+	    tr ' ' '\n' |					\
+	    sed -E 's/.c$/.o/' )
+	CPP_ARGS_FIXED="-std=c99 -DCPUSUPPORT_CONFIG_FILE=\"cpusupport-config.h\" -I${SUBDIR_DEPTH} -MM"
+	OUT_CC_BEGIN="\${CC} \${CFLAGS_POSIX} ${CFLAGS_HARDCODED}"
+	OUT_CC_MID="-I${SUBDIR_DEPTH} \${IDIRS} \${CPPFLAGS} \${CFLAGS}"
+
+	# Generate build instructions for each object
+	for F in $OBJ; do
+		S=`${MAKEBSD} source-${F}`
+		CF=`${MAKEBSD} cflags-${F}`
+		IDIRS=`${MAKEBSD} -V IDIRS`
+		echo `${CPP} ${S} ${CPP_ARGS_FIXED} ${IDIRS} -MT ${F}` | \
```

Comment:
I don't understand why this is
```
echo `foo` | bar
```
rather than just
```
foo | bar
```
?

Reply:
That's left-over from what you wrote in `Makefile.BSD`, so don't look at me!  :P

Reply:
hmm.  Adding the `echo` combines all the lines together -- the `${CPP}` helpfully wraps lines.  Granted, there's other ways of combining multiples lines in a posix sh.

Reply:
Looks to me like you introduced that here:
https://github.com/Tarsnap/spiped/commit/b99a843e40a27afd20cf36638a8f533554d8dbf7#diff-e541a5d9bd0f63955b28217432f983eb

;-)

I don't know, maybe using `echo` is cleaning up the whitespace?  Try changing it and see if things break.

Reply:
Ok.  In a REBASE commit, I've replaced:
- 1 `echo` and 1 `sed`

with:
- 2 `sed`s and 2 `tr`s.

Umm, enjoy?

Reply:
Yeah, I guess it's a bit ugly.  But I still think this is better than "subshell inside echo command"